pilot/pkg/networking/core/listener_builder.go

```diff
@@ -307,9 +308,29 @@ func (lb *ListenerBuilder) buildHTTPConnectionManager(httpOpts *httpListenerOpts
 	if httpOpts.http3Only {
 		connectionManager.CodecType = hcm.HttpConnectionManager_HTTP3
 		connectionManager.Http3ProtocolOptions = &core.Http3ProtocolOptions{}
+	} else if lb.push.Mesh.GetDefaultConfig().GetProxyHeaders().GetPreserveHttp1HeaderCase() {
```

Comment:
We don't want to only check default config; there may be a proxyconfig annotation applied to this particular proxy (e.g. deployment). Use `lb.push.ProxyConfigs.EffectiveProxyConfig()` instead

Reply:
Is this the only other config?

Reply:
Yes, its the combination of all `ProxyConfig` settings across `meshConfig`, annotations and the CRDs